src/main/utility/counter.rs

```diff
@@ -80,6 +151,24 @@ impl Display for Counter {
     }
 }
 
+impl Add for Counter {
+    type Output = Self;
+    /// Combines two counters by adding all values for all keys of `other` to `self`.
+    fn add(mut self, other: Self) -> Self {
+        self.add_counter(&other);
+        self
+    }
+}
+
+impl Sub for Counter {
+    type Output = Self;
+    /// Combines two counters by subtracting all values for all keys of `other` from `self`.
+    fn sub(mut self, other: Self) -> Self {
+        self.sub_counter(&other);
+        self
+    }
+}
```

Comment:
Does it make sense to subtract counters? For example if counter `a` has keys that counter `b` doesn't, then I think `assert_eq!(b-a+a, b)` would fail, which might be unexpected.

Reply:
I think subtracting makes sense, but subtracting 1-5 to equal 0 is weird. I'll change the value type to `i64` instead of `u64` to support negative values too.